google-cloud-apis/java-apis/maven-apis/src/com/google/cloud/tools/intellij/cloudapis/maven/MavenCloudApiActionDecorator.java

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.cloudapis.maven;
+
+import com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacet;
+import com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetType;
+import com.google.cloud.tools.intellij.appengine.java.maven.project.MavenProjectService;
+import com.google.cloud.tools.intellij.appengine.java.project.AppEngineProjectService;
+import com.google.cloud.tools.intellij.cloudapis.CloudApiActionDecoratorExtension;
+import com.google.cloud.tools.intellij.cloudapis.GoogleCloudApisMessageBundle;
+import com.intellij.facet.FacetManager;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.JavaVersion;
+
+/**
+ * {@link CloudApiActionDecoratorExtension} extension point that checks for Java projects version if
+ * App Engine is used, and ensures Maven dependencies support is only available for Maven projects.
+ * Otherwise, action is disabled and help text is shown.
+ */
+public class MavenCloudApiActionDecorator implements CloudApiActionDecoratorExtension {
+
+  @Override
+  public void decorate(AnActionEvent e) {
+    boolean addLibrariesEnabled = false;
+    if (e.getProject() != null) {
+      Set<CloudLibrariesModuleSupportType> moduleSupportTypes =
+          Stream.of(ModuleManager.getInstance(e.getProject()).getModules())
+              .map(this::checkModuleForAddCloudLibraries)
+              .collect(Collectors.toSet());
+
+      addLibrariesEnabled = moduleSupportTypes.contains(CloudLibrariesModuleSupportType.SUPPORTED);
+      if (!addLibrariesEnabled) {
+        // update message to hint what is missing.
+        for (CloudLibrariesModuleSupportType supportType : moduleSupportTypes) {
+          switch (supportType) {
+            case MAVEN_REQUIRED:
+              e.getPresentation()
+                  .setDescription(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.maven.required.description"));
+              e.getPresentation()
+                  .setText(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.disabled.maven.text"));
+              break;
+            case APPENGINE_JAVA8_REQUIRED:
+              e.getPresentation()
+                  .setDescription(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.gae.java8.required.description"));
+              e.getPresentation()
+                  .setText(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.disabled.java8.text"));
+              break;
```

Comment:
nit: the standards doc suggest to always include a default case. In this case, since you don't need to populate it, it can have a comment to indicate why you are skipping / what happens in the default case.